src/js/SelectionControls/SelectionControl.d.ts

```diff
@@ -15,6 +17,10 @@ export interface BaseSelectionControlProps extends Props {
   inline?: boolean;
   'aria-label'?: string;
   'aria-labelledby'?: IdPropType;
+  controlClassName: ClassNameType;
```

Comment:
Can you update these new props to be optional and update the base `className` to use the new type?

As a side note, I'm not sure I like the pattern of using a `className` function for this, I think it would be better to make the components controlled if you need to dynamically change the `className`. I think it'll be fine for here since it's backwards compatible and there are a couple of other components that do this I think, but I'm not sure if I'll follow this pattern for v2.

Reply:
> Can you update these new props to be optional

It was my oversight. I'll update types.

> and update the base className to use the new type?

I tried to add `className` into `BaseSelectionControlProps` but got an error from TypeScript compiler. I'll try one more time.

>  I'm not sure I like the pattern of using a className function for this, I think it would be better to make the components controlled if you need to dynamically change the className. ...  I'm not sure if I'll follow this pattern for v2

I've added support for functional `className` to maximize components flexibility. But I understand your concerns. If you would like I can remove support for functional prop values to be forward compatible with v2.

Reply:
Ahh sorry, I should have really made that comment in my notes instead of a here. It was more a reminder to me. I think it makes a lot of sense here and adds the extra flexibility needed. So keep it here :) 

I'm redoing the component api/patterns a bit in v2 to try to fix some of the complexity and un-flexibility of the more complex components and hopefully the flexibility provided here is also provided with the new api; just in a slightly different way.